services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if (!event) {
+    return ERR_CODE_INVALID_QUEUE_MSG;
+  } else if (!thermalMgrQueueHandle) {
+    return ERR_CODE_INVALID_STATE;
+  }
+  BaseType_t errCode = xQueueSend(thermalMgrQueueHandle, event, 0);
+  if (errCode == errQUEUE_FULL) {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt = { THERMAL_MGR_EVENT_OS_INTERRUPT };
+  error_code_t errCode = thermalMgrSendEvent(&interrupt);
+  if (errCode) {
+    printConsole("%d\n", errCode);
+  }
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
+    thermal_mgr_event_t currentItem;
+    BaseType_t received = xQueueReceive(thermalMgrQueueHandle, &currentItem, 5000 / portTICK_PERIOD_MS);
     
+    if (received == pdTRUE) {
+      float currentTemp = 0.0;
+      uint8_t devAddr = ((lm75bd_config_t *)pvParameters)->devAddr;
+      error_code_t errCode = readTempLM75BD(devAddr, &currentTemp);
```

Comment:
check the event type before reading temp since if it is something other than THERMAL_MGR_EVENT_MEASURE_TEMP_CMD or THERMAL_MGR_EVENT_OS_INTERRUPT like if it was invalid we should not be reading the temp